app/src/main/assets/native/injectionScript.js

```diff
@@ -0,0 +1,16 @@
+!function() {
+    var scripts = [
+        '/native/nativeshell.js',
+        '/native/EventEmitter.js',
+        document.currentScript.src.concat('?', Date.now())
+    ];
+    scripts.forEach(function(src) {
+        var scriptElement = document.createElement('script');
+        scriptElement.type = 'text/javascript';
+        scriptElement.src = src;
+        scriptElement.charset = 'utf-8';
+        scriptElement.setAttribute('defer', '');
+        document.body.appendChild(scriptElement);
+    });
+    document.currentScript.remove();
+}();
```

Comment:
Made some changes  to the script, mainly using ES6 syntax since the webview requires it and it makes the code slightly easier to read. I also moved the cache buster so we're 100% sure the injected scripts are not outdated (although I think that's not possible anyway). Also moved the script elements to the head instead of body.

```suggestion
(() => {
    const scripts = [
        '/native/nativeshell.js',
        '/native/EventEmitter.js',
        document.currentScript.src
    ];

    for (const script of scripts) {
        const scriptElement = document.createElement('script');
        scriptElement.src = script.concat('?', Date.now());
        scriptElement.charset = 'utf-8';
        scriptElement.setAttribute('defer', '');

        document.head.appendChild(scriptElement);
    }
})();
```

Reply:
> I also moved the cache buster so we're 100% sure the injected scripts are not outdated (although I think that's not possible anyway).

Yes, it is definitely not possible, this is because all intercepted resources are never cached.

![image](https://user-images.githubusercontent.com/25919231/122556039-a747a580-d000-11eb-9ae3-c9a22b655c91.png)

While the rest of the resources are cached

![image](https://user-images.githubusercontent.com/25919231/122556670-67cd8900-d001-11eb-9334-a5b1cb9a9770.png)

So I'm going to leave it as it was to make the logcat slightly easier to read